### PR TITLE
Legion: add Rust-based profiler to install

### PR DESCRIPTION
**Pull request by rbberger**
This adds a `prof` variant which when enabled installs the Rust-based Legion prof.

@elliottslaughter let me know what you think.

### CODE REVIEW

File: var/spack/repos/builtin/packages/legion/package.py [hunk 1]
```diff
@@ -277,6 +279,9 @@ def validate_gasnet_root(value):
         default=1024,
         description="Maximum number of nodes supported by Legion.",
     )
+    variant("prof", default=False, description="Install Rust Legion prof")
+
+    depends_on("rust", type="build", when="+prof")
```
Comment (elliottslaughter) [inline var/spack/repos/builtin/packages/legion/package.py]: Our profiler depends on at least Rust 1.74. Is that something we can assume here?

**Comment by elliottslaughter** [comment]
This seems reasonable to me, other than possibly specifying a minimum Rust version.

File: var/spack/repos/builtin/packages/legion/package.py (hunk 1 above)
Comment (rbberger) [reply var/spack/repos/builtin/packages/legion/package.py]: I can add that, currently it supports up to 1.76

Comment (rbberger) [reply var/spack/repos/builtin/packages/legion/package.py]: Updated.

**Review by elliottslaughter — APPROVED** [review]
Looks good to me.

**Comment by rbberger** [comment]
@tldahlgren could you please look at this?

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**